packages/shared/test/types.test.ts

```diff
@@ -0,0 +1,508 @@
+import { z } from 'zod';
+
+import { zSanitizedJustification } from '../src/types/7715-permissions-types';
+
+describe('zSanitizedJustification', () => {
+  describe('Basic validation', () => {
+    it('should reject empty strings', () => {
```

Comment:
nit: (plus this is certainly not adhered to throughout, but I'm trying to follow for new tests) avoid "should" at the start of a test name, as this adds no value and makes the test names more monotonous. Names like `it('rejects empty strings')` still read naturally, and are more direct.